Adds a new option to skip socket lb when in pod ns (#13882)
cilium-team/cilium: Documentation/gettingstarted/kubeproxy-free.rst

.. _XDP acceleration:

Socket LoadBalancer Bypass in Pod Namespace
*******************************************

Cilium has built-in support for bypassing the socket-level LoadBalancer and falling back

> to the tc LoadBalancer when a custom redirection/operation relies on the original ClusterIP

Reviewer: What is a `tc LoadBalancer`? Maybe clarify here if easily possible.

Author: Added "at the veth interface" to help clarify, PTAL :)

within pod namespace (e.g., Istio side-car).

Setting ``loadBalancer.hostNamespaceOnly=true`` enables this bypassing mode. When enabled,
this circumvents socket rewrite in the ``connect()`` and ``sendmsg()`` syscall bpf hook and
will pass the original packet to next stage of operation (e.g., stack in
``per-endpoint-routing`` mode) and re-enables service lookup in the tc bpf program.

A Helm example configuration in a kube-proxy-free environment with socket LB bypass
looks as follows:

.. parsed-literal::

    helm install cilium |CHART_RELEASE| \\
        --namespace kube-system \\
        --set tunnel=disabled \\
        --set autoDirectNodeRoutes=true \\

>         --set kubeProxyReplacement=strick \\

Reviewer 2: typo sneaked in here, shouldn't it be `strict`

Author: Done.

.. _XDP acceleration:

Socket LoadBalancer Bypass in Pod Namespace
*******************************************

> Cilium has built-in support for bypassing the socket level LoadBalancer and falling back

Reviewer 3: ```suggestion
Cilium has built-in support for bypassing the socket-level LoadBalancer and falling back
```

Author: Done.

> to the tc LoadBalancer for the case where a custom redirection/operation relies on the

Reviewer 3: ```suggestion
to the tc LoadBalancer when a custom redirection/operation relies on the
```

Author: Done.

within pod namespace (e.g., Istio side-car).

> Setting ``loadBalancer.hostNamespaceOnly`` allows to enable this bypassing mode when set
> to ``true``. When enabled, this circumvents socket rewrite in the ``connect()`` syscall bpf

Reviewer 3: ```suggestion
Setting ``loadBalancer.hostNamespaceOnly=true`` enables this bypassing mode. When
enabled, this circumvents socket rewrite in the ``connect()`` syscall bpf
```

Author: Done.

> to ``true``. When enabled, this circumvents socket rewrite in the ``connect()`` syscall bpf

Reviewer 4: > When enabled, this circumvents socket rewrite in the ``connect()``

I think this is not complete - `__sock{4,6}_xlate_fwd()` is invoked by the `sock{4,6}_sendmsg()` too. So, it applies to the `sendmsg()` syscall too which can be used for non-connected UDP.

Author: ah makes sense, added `sendmsg()`.

> to ``true``. When enabled, this circumvent socket rewrite in the ``connect()`` syscall bpf

Reviewer 3: ```suggestion
to ``true``. When enabled, this circumvents socket rewrite in the ``connect()`` syscall bpf
```

Author: Done.

hook and will pass the original packet to next stage of operation (e.g., stack in
``per-endpoint-routing`` mode) and re-enables service lookup in the tc bpf program.

A Helm example configuration in a kube-proxy-free environment with socket LB bypass

> would look as follows:

Reviewer 3: ```suggestion
looks as follows:
```

Author: Done.
